### PR TITLE
Issue 5621: Depth/stencil render buffer fix on macOS/iOS

### LINKED ISSUE #5621: [OpenGL 3 - 1.2.180] gl error 36061: GL_FRAMEBUFFER_UNSUPPORTED
**Describe the bug (REQUIRED)**
`render.render_target` cause a crash on version 1.2.180. Not happens on 1.2.179.

Script:

```lua
self.rt = render.render_target("lowrez", {
		[render.BUFFER_COLOR_BIT] = color_params,
		[render.BUFFER_DEPTH_BIT] = depth_params,
		[render.BUFFER_STENCIL_BIT] = stencil_params
	})
```

Error:

```
ERROR:GRAPHICS: gl error 36061: GL_FRAMEBUFFER_UNSUPPORTED
Assertion failed: (false), function OpenGLNewRenderTarget, file ../src/opengl/graphics_opengl.cpp, line 2013.

ERROR:CRASH: CALL STACK:

# 0 pc     0xd055fd libxslt.1.dylib _sigtramp+29
Lua Callstack:
  /lowrezjam/render/lowrez.render_script:41: in function setup
  /lowrezjam/render/lowrez.render_script:59: in function </lowrezjam/render/lowrez.render_script:48>
```

**Defold version (REQUIRED):**
  - Version 1.2.180

**Platforms (REQUIRED):**
 - Platforms: macOS
 - OS: 10.15.7

**Minimal repro case project (OPTIONAL):**
[lowrezjam-template](https://github.com/britzl/lowrezjam-template)

**Logs (OPTIONAL):**
[_crash.zip](https://github.com/defold/defold/files/6089071/_crash.zip)

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 1]
```diff
@@ -1008,19 +1008,28 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
...
+
         if ((IsExtensionSupported("GL_OES_packed_depth_stencil", extensions)) || (IsExtensionSupported("GL_EXT_packed_depth_stencil", extensions)))
         {
             context->m_PackedDepthStencil = 1;
         }
         GLint depth_buffer_bits;
         glGetIntegerv( GL_DEPTH_BITS, &depth_buffer_bits );
+        if (GL_INVALID_ENUM == glGetError())
+        {
+            depth_buffer_bits = 24;
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:1023]: If this enum isn't supported, then the variable was uninitialized.

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 2]
```diff
@@ -1008,19 +1008,28 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
         context->m_DepthBufferBits = 16;
 #else
 
+#if defined(__MACH__)
+        context->m_PackedDepthStencil = 1;
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:1012]: Tested on macOS and iOS.

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 3]
```diff
@@ -1894,7 +1903,7 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
             glBindRenderbuffer(GL_RENDERBUFFER, rt->m_DepthStencilBuffer);
 #ifdef GL_DEPTH_STENCIL_ATTACHMENT
             // if we have the capability of GL_DEPTH_STENCIL_ATTACHMENT, create a single combined depth-stencil buffer
-            glRenderbufferStorage(GL_RENDERBUFFER, GL_DEPTH_STENCIL, rt->m_BufferTextureParams[param_buffer_index].m_Width, rt->m_BufferTextureParams[param_buffer_index].m_Height);
+            glRenderbufferStorage(GL_RENDERBUFFER, DMGRAPHICS_RENDER_BUFFER_FORMAT_DEPTH_STENCIL, rt->m_BufferTextureParams[param_buffer_index].m_Width, rt->m_BufferTextureParams[param_buffer_index].m_Height);
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:1906]: Depending on platform, it's defined as either `GL_DEPTH_STENCIL` or `GL_DEPTH24_STENCIL8_OES` (if `GL_DEPTH_STENCIL_OES` is defined)

**Review by Jhonnyg — APPROVED** [review]
👍

**Pull request merged**